src/olympia/devhub/views.py

```diff
@@ -580,10 +580,10 @@ def handle_upload(filedata, request, channel, app_id=None, version_id=None,
                 user_agent=request.META.get('HTTP_USER_AGENT'),
                 referrer=request.META.get('HTTP_REFERER'),
                 upload=upload)
-            akismet_checks = comment_check.si(
-                [report.id for report in akismet_reports])
         else:
-            akismet_checks = None
+            akismet_reports = []
+        akismet_checks = comment_check.si(
```

Comment:
you'll probably don't want to do that if `akismet_reports` is empty I guess?

Reply:
That's the part you _have_ to do to fix this.  No task means the wrong number of parameters are passed on to `validate_file_path` or `validate_file`.  `chain()` doesn't do this.  (`group()`, which makes a chord, does in eager mode but seemingly not in non-eager)

Reply:
Let's add a very verbose and explaining comment why this is necessary, it broke before and I fear if we touch it again in a few months time we are doomed to do that again. Both here, and in `devhub.tasks` given that this is generic functionality that's potentially exposed to every `validation()` caller.